### PR TITLE
[misc] Output "taichi_cpp_tests.exe" to the "bin" folder on Windows

### PR DESCRIPTION
**Pull request by xumingkuan**
Related issue = #2502

By default, on Windows, Taichi outputs `taichi_cpp_tests.exe` in somewhere like `build\RelWithDebInfo\`, which is hard to find.

This PR moves it to `bin\`, which is in the `PATH` and we can run `taichi_cpp_tests` everywhere.

### CODE REVIEW

File: docs/lang/articles/contribution/writing_cpp_tests.md [hunk 1]
```diff
@@ -21,6 +21,11 @@ make
 ./taichi_cpp_tests
 ```
 
+:::note
+On Windows, `taichi_cpp_tests.exe` will be outputted into the `%TAICHI_REPO_DIR%\bin` directory,
+so you can directly run it after adding the directory to `PATH` in [Setting up Taichi for development](dev_install.md#setting-up-taichi-for-development-3).
```
Comment (xumingkuan) [inline docs/lang/articles/contribution/writing_cpp_tests.md:26]: Hopefully I understand how the link works correctly -- feel free to correct me if the syntax is wrong.

File: docs/lang/articles/contribution/writing_cpp_tests.md [hunk 2]
```diff
@@ -21,6 +21,11 @@ make
 ./taichi_cpp_tests
 ```
 
+:::note
+On Windows, `taichi_cpp_tests.exe` will be outputted into the `%TAICHI_REPO_DIR%\bin` directory,
```
Comment (k-ye) [inline docs/lang/articles/contribution/writing_cpp_tests.md:25]:
```suggestion
On Windows, `taichi_cpp_tests.exe` will be placed inside the `%TAICHI_REPO_DIR%\bin` directory,
```

File: cmake/TaichiTests.cmake [hunk 3]
```diff
@@ -23,6 +23,14 @@ include_directories(
 )
 
 add_executable(${TESTS_NAME} ${TAICHI_TESTS_SOURCE})
+if (WIN32)
+    # Output the executable to bin/ instead of build/Debug/...
+    set_target_properties(${TESTS_NAME} PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/bin")
```
Comment (k-ye) [inline cmake/TaichiTests.cmake:28]: nit: Shall we create a variable for this?

**Review by k-ye — APPROVED** [review]
Thanks!

File: cmake/TaichiTests.cmake (hunk 3 above)
Comment (xumingkuan) [reply cmake/TaichiTests.cmake:28]: A variable for the output place or what?

Comment (k-ye) [reply cmake/TaichiTests.cmake:28]: right, for the output place

**Review by k-ye — APPROVED** [review]
LGTM!

**Pull request merged**